### PR TITLE
Make haproxy maxconn configurable

### PR DESCRIPTION
**Pull request by pecameron**
bug fix in 3.5 

The haproxy maxconn (maximum connections) is by default 2000. This
change make that configurable with a defaul tof 20000 through the
oadm router --max-connections= option when creating a router and through
changing the ROUTER_MAX_CONNECTIONS environment variable in an existing
router.

origin PR 12716

bug 1405440
https://bugzilla.redhat.com/show_bug.cgi?id=1405440

### CODE REVIEW

**Comment by pecameron** [comment]
@knobunc PTAL

File: install_config/router/default_haproxy_router.adoc [hunk 1]
```diff
@@ -207,6 +207,28 @@ $ oadm router region-west -o yaml --images=myrepo/somerouter:mytag \
...
+health check from working which causes unneeded restarts. The system needs to be configured
+to support the number of connections.  The limits shown in 'sysctl fs.nr_open' and 'sysctl fs.file-max'
+must be large enough or haproxy will not start.
+
+
+When the router is created the --max-connections= options sets the desired limit, 
+
+====
+----
+$ oadm router ----max-connections=10000   ....
```
Comment (juanvallejo) [inline install_config/router/default_haproxy_router.adoc]: only two dashes before the flag: `--max-connections=10000`

File: install_config/router/default_haproxy_router.adoc [hunk 2]
```diff
@@ -207,6 +207,28 @@ $ oadm router region-west -o yaml --images=myrepo/somerouter:mytag \
...
+Setting the Maximum Number of Concurrent Connections::
+
+The default maximum number of connections that can be handled by the router is 2000. 
+You can change that limit depending on your needs.  Too few connections will prevent the
+health check from working which causes unneeded restarts. The system needs to be configured
+to support the number of connections.  The limits shown in 'sysctl fs.nr_open' and 'sysctl fs.file-max'
+must be large enough or haproxy will not start.
+
+
+When the router is created the --max-connections= options sets the desired limit,
```
Comment (juanvallejo) [inline install_config/router/default_haproxy_router.adoc]: nit: s/options/option

File: install_config/router/default_haproxy_router.adoc [hunk 3]
```diff
@@ -207,6 +207,28 @@ $ oadm router region-west -o yaml --images=myrepo/somerouter:mytag \
 ====
 endif::[]
 
+[[concurrent-connections]]
+Setting the Maximum Number of Concurrent Connections::
+
+The default maximum number of connections that can be handled by the router is 2000. 
+You can change that limit depending on your needs.  Too few connections will prevent the
+health check from working which causes unneeded restarts. The system needs to be configured
+to support the number of connections.  The limits shown in 'sysctl fs.nr_open' and 'sysctl fs.file-max'
```
Comment (juanvallejo) [inline install_config/router/default_haproxy_router.adoc]: `The system needs to be configured to support the number of connections needed by the health check`

Comment (pecameron) [inline install_config/router/default_haproxy_router.adoc]: Well, not exactly. max-connections is the size of the pool. healthz and users all take from the pool, so there need to be enough satisfy all the users of the pool. This is fuzzy and must "big enough". Sigh.

**Comment by pecameron** [comment]
@juanvallejo Thanks for reviewing this.

File: install_config/router/default_haproxy_router.adoc (hunk 3 above)
Comment (juanvallejo) [inline install_config/router/default_haproxy_router.adoc]: Thanks for clarifying

**Review by knobunc — APPROVED** [review]
LGTM.  @openshift/team-documentation over to you

**Comment by pecameron** [comment]
@openshift/team-documentation Anything else needed here? PTAL

**Comment by ahardin-rh** [comment]
Thank you! I will make some very minor style edits in a follow-up PR.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Added `ROUTER_MAX_CONNECTIONS` to the xref:../architecture/core_concepts/routes.adoc#env-variables[Router Configuration Parameters] table.
%
|xref:../install_config/router/default_haproxy_router.adoc#install-config-router-default-haproxy[Setting up a Router -> Using the Default HAProxy Router]
|Added a new section about xref:../install_config/router/default_haproxy_router.adoc#concurrent-connections[Setting the Maximum Number of Concurrent Connections].
%

**Pull request merged**